baseservice/iodevices/base/input_devices.py

```diff
@@ -60,20 +65,31 @@ def read_message(self,
         :return: a tuple of (Message, DeviceHeaders, Transaction) or EMPTY_RESULT if no message was available.
         the device headers, can contain extra information about the device that returned the message
         """
-        message, device_headers, transaction = self._read_message(timeout=timeout,
-                                                                  with_transaction=with_transaction)
-        if message is not None:
+        result = self._read_message(timeout=timeout, with_transaction=with_transaction)
+        if self.is_non_empty_message_result(result):
```

Comment:
o.k. we really need to find an alternative for this...
you must explain why is this like that.